src/System.Diagnostics.Process/tests/System.Diagnostics.Process.Tests/ProcessTest.cs

```diff
@@ -3,580 +3,381 @@
-            TimeSpan elapsedTime = new TimeSpan(dt2.Ticks - exitTime.Ticks);
-            Assert.True(elapsedTime.Seconds <= 1, "Process_ExitTime is incorrect.");
+            StartAndKillProcessWithDelay(p);
+            Assert.True(p.ExitTime < DateTime.Now, "Process_ExitTime is incorrect.");
         }
 
 
-
         [Fact]
-        public static void Process_GetHandle()
+        public void Process_GetHandle()
         {
-            Process p = CreateProcessInfinite();
-            p.Start();
-            try
-            {
-                Assert.Equal(p.Id, Interop.GetProcessId(p.SafeHandle));
-            }
-            finally
-            {
-                p.Kill();
-                p.WaitForExit();
-            }
+            Assert.Equal(s_process.Id, Interop.GetProcessId(s_process.SafeHandle));
         }
 
         [Fact]
-        public static void Process_HasExited()
+        public void Process_HasExited()
         {
-            Process p = CreateProcess();
-            p.Start();
-            p.WaitForExit();
-            Assert.True(p.HasExited, "Process_HasExited001 failed");
-
-            p = CreateProcessInfinite();
-            p.Start();
-            try
             {
-                Assert.False(p.HasExited, "Process_HasExited002 failed");
+                Process p = CreateProcess();
+                p.Start();
+                DelayTask();
+                Assert.True(p.HasExited, "Process_HasExited001 failed");
             }
-            finally
+
             {
-                p.Kill();
-                p.WaitForExit();
+                Process p = CreateProcessInfinite();
+                p.Start();
+                try
+                {
+                    Assert.False(p.HasExited, "Process_HasExited002 failed");
+                }
+                finally
+                {
+                    p.Kill();
+                    p.WaitForExit();
+                }
+
+                Assert.True(p.HasExited, "Process_HasExited003 failed");
             }
-            
-            Assert.True(p.HasExited, "Process_HasExited003 failed");
         }
 
         [Fact]
-        public static void Process_MachineName()
+        public void Process_MachineName()
         {
             // Checking that the MachineName returns some value.
-            Process p = CreateProcessInfinite();
-            p.Start();
-            try
-            {
-                Console.WriteLine(p.MachineName);
-            }
-            finally
-            {
-                p.Kill();
-                p.WaitForExit();
-            }
+            Assert.NotNull(s_process.MachineName);
         }
 
         [Fact]
-        public static void Process_MainModule()
+        public void Process_MainModule()
         {
             // Get MainModule property from a Process object
-            Process p = CreateProcessInfinite();
-            p.Start();
-            try
+            string moduleName = s_process.MainModule.ModuleName;
```

Comment:
The open issue related to MainModule shows that ModuleName can internally throw a null reference exception.  MainModule looks like this today:
```
ModuleInfo module = NtProcessManager.GetFirstModuleInfo(_processId);
return new ProcessModule(module);
```
If GetFirstModuleInfo returns null (which is possible), module will be null, and ProcessModule will be constructed wrapping that null value.  ProcessModule.ModuleName looks like this:
```
public string ModuleName
{
    get { return _moduleInfo._baseName; }
}
```
so if the ProcessModule is constructed to wrap a null value, _moduleInfo will be null and this will dereference a null reference.

In other words, there's a real product bug of some sort here, and even if updating this test hides that, we shouldn't lose sight of there being an actual issue to fix.